api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -327,12 +358,13 @@ func (m *AzureManagedControlPlane) validateDisableLocalAccounts(_ client.Client)
 }
 
 // validateVersion validates the Kubernetes version.
-func (m *AzureManagedControlPlane) validateVersion(_ client.Client) error {
-	if !kubeSemver.MatchString(m.Spec.Version) {
-		return errors.New("must be a valid semantic version")
+func validateVersion(version string, fldPath *field.Path) field.ErrorList {
```

Comment:
Refactored these webhook validation functions to be usable by the template types